### PR TITLE
Calendar backend week 5

### PR DESCRIPTION
**Pull request by gabbygermanson**

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/TimeRange.java [hunk 1]
```diff
@@ -54,61 +33,63 @@ private TimeRange(int start, int duration) {
...
     return start + duration;
   }
 
-  /**
-   * Checks if two ranges overlap. This means that at least some part of one range falls within the
-   * bounds of another range.
-   */
-  public boolean overlaps(TimeRange other) {
+
+  /**A comparator for sorting ranges by their START time in ascending order.*/
```
Comment (theannielin) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/TimeRange.java:52]: Nit: Space after /**

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/TimeRange.java [hunk 2]
```diff
@@ -54,61 +33,63 @@ private TimeRange(int start, int duration) {
...
+  public static final Comparator<TimeRange> ORDER_BY_START = new Comparator<TimeRange>() {
+    @Override
+    public int compare(TimeRange a, TimeRange b) {
+      return Long.compare(a.start, b.start);
+    }
+  };
+
+  /**A comparator for sorting ranges by their END time in ascending order.*/
```
Comment (theannielin) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/TimeRange.java:60]: Same nit as above

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/TimeRange.java [hunk 3]
```diff
@@ -120,66 +101,63 @@ public boolean contains(int point) {
...
+    return Long.hashCode(start) ^ Long.hashCode(duration);
+  }
+
+
+  @Override
+  public String toString() {
+    return String.format("Range: [%d, %d)", start, start + duration);
+  }
+
+
```
Comment (theannielin) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/TimeRange.java:141]: Some extra new lines in this file

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 4]
```diff
@@ -14,10 +14,147 @@
...
+            return timesWithOptionals;
+        }
+
+        Collection<String> requestersWithOptionals = new ArrayList<>();
+        requestersWithOptionals.addAll(request.getAttendees());
+        requestersWithOptionals.addAll(request.getOptionalAttendees());
+        Collection<String> requestersNoOptionals = new ArrayList<>();
+        requestersNoOptionals.addAll(request.getAttendees());
+
+        ArrayList<Event> eventsWithOptionals = new ArrayList<>(events);
```
Comment (theannielin) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:45]: This is a rather large chunk of code and hard to read. Could you separate this out into different helper method that address the different cases?

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 5]
```diff
@@ -14,10 +14,147 @@
...
+                if (!relevantEvent) {
+                    currEventList.remove(currEvent);
+                    eventsIndex--;
+                    eventsLength--;
+                }
+                
+            }
+        }
+
+        ArrayList<Event> eventsList = eventsWithOptionals;
```
Comment (theannielin) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:76]: I think the event parsing portion of this code and the time parsing portion should be in separate helper methods.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 4 above)
Comment (gabbygermanson) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Thank you for this comment, I totally forgot about that best coding practice and as you saw was doing for loops for two iterations

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 6]
```diff
@@ -23,138 +23,151 @@
...
 public final class FindMeetingQuery {
-    
-    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-        ArrayList<TimeRange> timesWithOptionals = new ArrayList<>();
-        ArrayList<TimeRange> timesNoOptionals = new ArrayList<>();
-        timesWithOptionals.add(TimeRange.WHOLE_DAY);
-        timesNoOptionals.add(TimeRange.WHOLE_DAY);
 
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
         if (request.getDuration() > (24 * 60)) {
```
Comment (theannielin) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:28]: Nit: Make this a contant

**Pull request merged**